### ISSUE TITLE
Sylius\Bundle\SearchBundle\Accessor\PropertyAccessor doesn't do anything :)

### ISSUE THREAD

**Issue by pamil** [Sylius/Sylius#3211]
`Sylius\Bundle\SearchBundle\Accessor\PropertyAccessor` is badly written (it creates property accessors based on FQCN, then iterates through only the first one (returns its value or returns null if there's no such property) and then has dead code block.

It is only used by `sylius.search.transformers.model.product` service (`FOS\ElasticaBundle\Transformer\ModelToElasticaAutoTransformer`) as a proxy for `Sylius\Component\Product\Accessor\ProductAccessor`.

To sum it up, we can delete it and replace with `ProductAccessor` and application behaviour won't change, but less code is better code :)

What do you think?

**Comment by pjedrzejewski** [comment]
Let's do this! But I'd like to maybe try get feedback from @agounaris! What was the reason behind PropertyAccessor?
